spatialmath/box.go

```diff
@@ -352,3 +355,84 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// TODO: add function description
+func (b *box) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
+
+	dims := [3]float64{b.halfSize[0] * 2, b.halfSize[1] * 2, b.halfSize[2] * 2}
+
+	var faces [][]float64
+	// which faces are these
+	for i := 0.0; i <= dims[0]; i += options["xIter"].(float64) {
+		for k := 0.0; k <= dims[2]; k += options["zIter"].(float64) {
+			p1 := []float64{i, 0, k}
+			p2 := []float64{i, dims[1], k}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// which faces are these
+	for j := 0.0; j <= dims[1]; j += options["yIter"].(float64) {
+		for k := 0.0; k <= dims[2]; k += options["zIter"].(float64) {
+			p1 := []float64{0, j, k}
+			p2 := []float64{dims[0], j, k}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// which faces are these
+	for i := 0.0; i <= dims[0]; i += options["xIter"].(float64) {
+		for j := 0.0; j <= dims[1]; j += options["yIter"].(float64) {
+			p1 := []float64{i, j, 0}
+			p2 := []float64{i, j, dims[2]}
+			faces = append(faces, p1, p2)
+		}
+	}
+	// what does this do
+	min := b.Vertices()[0]
+	for i := 0; i < len(b.Vertices()); i++ {
+		curr := b.Vertices()[i]
+		if curr.X <= min.X && curr.Y <= min.Y && curr.Z <= min.Z {
+			min = b.Vertices()[i]
+		}
+	}
+	// what does this do
+	for _, v := range faces {
```

Comment:
I *think* that both this, and the matrix transformation below, can be replaced by transforming each point by `PoseInverse(box.Pose())`

Reply:
gotcha, I will try it out.

Reply:
I agree that this will work but this part of the code will be gone in the newest version - pushing now